uyuni-tools/sync_client_tools.py

```diff
@@ -0,0 +1,116 @@
+    products = client.sync.content.listProducts(key)
+    for product in products:
+        if product.get('status', '').lower() == 'installed':
+            product_name = product.get('friendly_name', 'Unknown Product')
+            extensions = product.get('extensions', [])
+            if extensions:
+                for ext in extensions:
+                    all_extensions.append(ext)
+            else:
+                print(f"No extensions found for {product_name}.")
+    return all_extensions
+
+def add_client_tools_channels(client, key, extensions, dry_run):
+    """Add all the client tools channels available which aren't synced yet"""
+    for ext in extensions:
+        if "Client Tools" in ext.get('friendly_name', ''):
+            channels = ext.get('channels', [])
+            if not channels:
+                print("  No channels found.")
+                continue
+            for ch in channels:
+                if ch.get('family') == 'SLE-M-T' and ch.get('optional') != True:
+                    if ch.get('status', '').lower() == 'installed':
+                        log(f"already synced: {ch.get('label')}")
+                        continue
+                    label = ch.get('label', 'N/A')
+                    family = ch.get('family', 'N/A')
+                    if dry_run:
+                        dry_run_log(f"Would add channel: {label}")
+                    else:
+                        log(f"Adding channel: {label}")
+                        client.sync.content.addChannel(key, label, '')
+
+def main():
+    """Main function to run the entire workflow."""
+    parser = argparse.ArgumentParser(description="A script to add client tools channels to SUSE Manager.",
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("--no-dry-run", action="store_true", 
+                        help="Perform actual changes instead of a dry run.\n"
+                             "The script runs in dry-run mode by default.")
+    args = parser.parse_args()
+    dry_run = not args.no_dry_run
+
+    if dry_run:
+        log("Script is running in DRY-RUN mode. No changes will be made.")
+    else:
+        log("WARNING: Script is running in LIVE mode. Changes will be applied.")
+
+    client, key = connect_and_login()
+    if not client:
+        return
+
+    try:
+        extensions = find_extensions_of_synced_products(client, key)
+        add_client_tools_channels(client, key, extensions, dry_run)
+        log("INFO: Channels have been added and will be synced with the next scheduled repository sync. If you want to sync them now, you can schedule the 'Single run schedule' task from the 'mgr-sync-refresh-bunch' family of tasks.")
+    
+    except Exception as e:
+        print(f"[ERROR] An unexpected error occurred: {e}")
+    finally:
+        if key:
+            client.auth.logout(key)
+            log("Logged out successfully.")
+
+if __name__ == "__main__":
+    main()
```

Comment:
```suggestion
    main()

```
A newline is missing at the end.